client/ayon_core/tools/publisher/control.py

```diff
@@ -340,27 +377,39 @@ class PublishPluginsProxy:
     """
 
     def __init__(self, plugins):
-        plugins_by_id = {}
-        actions_by_plugin_id = {}
-        action_ids_by_plugin_id = {}
+        self._plugins_by_id = {}
+        self._actions_by_plugin_id = {}
+        self._action_ids_by_plugin_id = {}
+
         for plugin in plugins:
             plugin_id = plugin.id
-            plugins_by_id[plugin_id] = plugin
+            self._plugins_by_id[plugin_id] = plugin
+
```

Comment:
I'm calling it after the plugins have been processed since it needs the results to retrieve the actions added during processing and any validation errors that play a role in filtering the actions.

Reply:
Okay, sounds good !